lib/ach_client/providers/sftp/ach_transaction.rb

```diff
@@ -26,13 +26,30 @@ def to_entry_detail
       end
 
       private
+
+      def amount_in_cents
+        # Take absolute value in case amount is negative
+        Helpers::DollarsToCents.dollars_to_cents(amount.abs)
+      end
+
       def transaction_code
         [
           AccountTypeTransformer.serialize_to_provider_value(
             account_type
           ),
           TransactionTypeTransformer.serialize_to_provider_value(
-            transaction_type
+            # Some NACHA providers can't handle negative amounts
+            # So we flip the amount and toggle the transaction type between
+            #   Debit and Credit
+            if amount.negative?
```

Comment:
you are absolutely right that my return statement within a set of args is awful lol.
Maybe we move it to a private method and encapsulate that functionality there? In that function you can rework the branch logic to reduce the chunkiness.